client/unfoldable/unfoldable.js

```diff
@@ -0,0 +1,233 @@
+
+function convertJson(data) {
+
+  let id = typeof(data.root) == 'string' ? data.root : 'root'
+  let processed = []
+  let depth = 0
+
+  function replaceObjectIdsByActualObjects (id, data, processed, depth) {
```

Comment:
nesting function declarations within function declarations is fine, but it may affect performance if you call the parent function several times since you will be re-declaring the same function several times.